code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/LoadBalancerServiceInstanceStopPreListener.java

```diff
@@ -0,0 +1,220 @@
+import io.cattle.platform.agent.RemoteAgent;
+import io.cattle.platform.async.utils.AsyncUtils;
+import io.cattle.platform.core.addon.LbConfig;
+import io.cattle.platform.core.addon.PortRule;
+import io.cattle.platform.core.constants.InstanceConstants;
+import io.cattle.platform.core.constants.ServiceConstants;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.dao.InstanceDao;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.core.model.Service;
+import io.cattle.platform.core.model.ServiceExposeMap;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.handler.ProcessPreListener;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.eventing.exception.EventExecutionException;
+import io.cattle.platform.eventing.model.Event;
+import io.cattle.platform.eventing.model.EventVO;
+import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.object.meta.ObjectMetaDataManager;
+import io.cattle.platform.object.util.DataAccessor;
+import io.cattle.platform.object.util.ObjectUtils;
+import io.cattle.platform.process.common.handler.AgentBasedProcessHandler;
+import io.cattle.platform.servicediscovery.api.util.ServiceDiscoveryUtil;
+import io.cattle.platform.servicediscovery.service.ServiceDiscoveryService;
+import io.cattle.platform.util.exception.ExecutionException;
+import io.cattle.platform.util.type.CollectionUtils;
+import io.cattle.platform.util.type.Priority;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+public class LoadBalancerServiceInstanceStopPreListener extends AgentBasedProcessHandler implements ProcessPreListener, Priority {
+
+    @Inject
+    JsonMapper jsonMapper;
+
+    @Inject
+    ServiceDiscoveryService sdService;
+
+    @Inject
+    InstanceDao instanceDao;
+
+    @Inject
+    GenericMapDao mapDao;
+
+    private static final Logger log = LoggerFactory.getLogger(LoadBalancerServiceInstanceStopPreListener.class);
+    private static final String LABEL_SERVICE_LAUNCH_CONFIG = "io.rancher.service.launch.config";
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        Instance instance = (Instance) state.getResource();
+        if (instance == null) {
+            return null;
+        }
+        // container.stop -> find container’s service -> find all the LB services of the account of the container
+        // service -> filter out those services having lbConfig.portRules having serviceId = container’s service
+        Service service = loadService(instance);
+        if (service == null) {
+            return null;
+        }
```

Comment:
We should disable draining by default. So once you get the service, find the drainTimeout and if it is 0 or null, do not drain.
This means also taking out the deafult value for the field in the json file you modified